codejail/jail.py

```diff
@@ -0,0 +1,298 @@
+"""
+Core Jail class and supporting code
+"""
+
+from collections import namedtuple
+import json
+import logging
+import os
+import shutil
+
+from .exceptions import JailError, SafeExecException
+from . import languages
+from . import limits
+from .proxy import run_subprocess_through_proxy
+from .subproc import run_subprocess
+from .util import json_safe, temp_directory
+
+log = logging.getLogger("codejail")
+
+
+JailResult = namedtuple('JailResult', ['status', 'stdout', 'stderr'])
+
+
+class Jail(object):
+    """
+    Self contained codejail context.
+    """
+    default_language = languages.other
+
+    known_languages = {
+        'python': languages.python2,
+    }
+
+    def __init__(self, command, bin_path, user, lang=None):
+        self.command = command
+        self.bin_path = bin_path
+        self.user = user
+        if lang is None:
+            self.lang = self.known_languages.get(command, self.default_language)
```

Comment:
Is it OK to silently use default_language if command is unknown? Seems like an error here would be better.  Maybe I don't understand how this object is created.  It could use more docstring.